source/Posets/InductiveTypesfromSmallGeneratedLattice.lagda

```diff
@@ -0,0 +1,1758 @@
+           → (U : Fam 𝓥 ⟨ L ⟩)
+           → ((order-of L) Joins.is-lub-of join-for L U) U holds
+is-lub-for (A , (_≤_ , ⋁_) , order , is-lub-of) = is-lub-of
+
+is-an-upper-bound-for_of_ : (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                          → (U : Fam 𝓥 ⟨ L ⟩)
+                          → ((order-of L) Joins.is-an-upper-bound-of
+                                          join-for L U) U holds
+is-an-upper-bound-for L of U = pr₁ (is-lub-for L U)
+
+is-least-upper-bound-for_of_ : (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                             → (U : Fam 𝓥 ⟨ L ⟩)
+                             → ((u' , _) : Joins.upper-bound (order-of L) U)
+                             → (order-of L (join-for L U) u') holds
+is-least-upper-bound-for L of U = pr₂ (is-lub-for L U)
+
+\end{code}
+
+We now define monotone endo-maps on lattice. This is sufficient for our work
+as we are studying fixed-points.
+
+\begin{code}
+
+module Monotone-Endo-Maps {𝓤 𝓦 𝓥 : Universe} (L : Sup-Lattice 𝓤 𝓦 𝓥) where
+
+ _≤_ : ⟨ L ⟩ → ⟨ L ⟩ → Ω 𝓦
+ _≤_ = order-of L
+
+ _is-monotone : (f : ⟨ L ⟩ → ⟨ L ⟩) → 𝓤 ⊔ 𝓦  ̇
+ f is-monotone = (x y : ⟨ L ⟩) → (x ≤ y) holds → (f x ≤ f y) holds
+
+\end{code}
+
+We now show that when one subset contains another the join of their total
+spaces are ordered as expected. This result will be quite useful throughout
+this file.
+
+\begin{code}
+
+module Subsets-Order-Joins {𝓤 𝓦 𝓥 : Universe}
+                           (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                           (A : 𝓥  ̇)
+                           (m : A → ⟨ L ⟩)
+                           where
+
+ _≤_ : ⟨ L ⟩ → ⟨ L ⟩ → Ω 𝓦
+ x ≤ y = order-of L x y
+
+ ⋁_ : Fam 𝓥 ⟨ L ⟩ → ⟨ L ⟩
+ ⋁_ = join-for L
+
+ open Joins _≤_
+
+ joins-preserve-containment : {P : 𝓟 {𝓥} A} {Q : 𝓟 {𝓥} A}
+                            → (C : P ⊆ Q)
+                            → ((⋁ ((Σ a ꞉ A , a ∈ P) , m ∘ pr₁))
+                            ≤ (⋁ ((Σ a ꞉ A , a ∈ Q ) , m ∘ pr₁))) holds
+ joins-preserve-containment {P} {Q} C =
+   (is-least-upper-bound-for L of ((Σ a ꞉ A , a ∈ P ) , m ∘ pr₁))
+    (⋁ ((Σ a ꞉ A , a ∈ Q ) , m ∘ pr₁) ,
+    λ (b , b-in-P)
+      → (is-an-upper-bound-for L of ((Σ a ꞉ A , a ∈ Q ) , m ∘ pr₁))
+    (b , C b b-in-P))
+
+\end{code}
+
+We now show if a type is small and has a map to the carrier then it has a join.
+This seems like strict requirement but as we will see it occurs often when
+considering a lattice with a basis. 
+
+\begin{code}
+
+module Small-Types-have-Joins {𝓤 𝓦 𝓥 𝓣 : Universe}
+                              (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                              (T : 𝓣  ̇)
+                              (m : T → ⟨ L ⟩)
+                              (t : T is 𝓥 small)
+                              where
+ 
+ _≤_ : ⟨ L ⟩ → ⟨ L ⟩ → Ω 𝓦
+ _≤_ = order-of L
+
+ ⋁_ : Fam 𝓥 ⟨ L ⟩ → ⟨ L ⟩
+ ⋁_ = join-for L
+
+ small-type : 𝓥  ̇
+ small-type = pr₁ t
+
+ small-≃ : small-type ≃ T
+ small-≃ = pr₂ t
+
+ small-map : small-type → T
+ small-map = ⌜ small-≃ ⌝
+
+ is-equiv-small-map : is-equiv small-map
```

Comment:
This breaks the naming conventions of `TypeTopology`. If we have  a map`f` and we want to state that `f` is an equivalence we write `f-is-equiv` and not `is-equiv-f` (names should make sense in English when we read them aloud). So this would give `T'-to-T-is-equiv` if you follow the above suggestion.